src/test-utils-new/extractor.ts

```diff
@@ -43,11 +43,11 @@ export default function extractDocumentation(
     if (extraExports.includes(className)) {
       continue;
     }
-    if (!(symbol.flags & ts.SymbolFlags.Class)) {
+    const classType = checker.getDeclaredTypeOfSymbol(symbol);
+    if (!classType.isClass()) {
       throw new Error(`Exported symbol is not a class, got ${checker.symbolToString(symbol)}`);
     }
 
-    const classType = checker.getTypeAtLocation(extractDeclaration(symbol));
```

Comment:
The actual fix. Replace `getTypeAtLocation` with `getDeclaredTypeOfSymbol`  to make typescript actually go cross-files to find the real type